thirdeye-persistence/src/main/java/ai/startree/thirdeye/datalayer/bao/NamespaceConfigurationManagerImpl.java

```diff
@@ -0,0 +1,195 @@
+    }
+    final Long id = dao.put(entity);
+    checkState(id != null, "failed to save namespace configuration entity");
+    entity.setId(id);
+    return id;
+  }
+
+  @Override
+  public int update(final NamespaceConfigurationDTO entity) {
+    return dao.update(entity);
+  }
+
+  @Override
+  public int update(final List<NamespaceConfigurationDTO> entities) {
+    return dao.update(entities);
+  }
+
+  @Override
+  public NamespaceConfigurationDTO findById(final Long id) {
+    return dao.get(id);
+  }
+
+  @Override
+  public @Nullable NamespaceConfigurationDTO findUniqueByNameAndNamespace(
+      final @NonNull String name, final @Nullable String namespace) {
+    final List<NamespaceConfigurationDTO> list = findByPredicate(
+        Predicate.AND(
+            Predicate.OR(
+                Predicate.EQ("namespace", namespace)
+            )
+        )
+    ).stream().filter(e -> Objects.equals(e.namespace(), namespace)).toList();
+
+    if (list.isEmpty()) {
+      return null;
+    } else if (list.size() == 1) {
+      return list.iterator().next();
+    } else {
+      throw new IllegalStateException(String.format(
+          "Found multiple entities with and namespace %s. This should not happen. Please reach out to support.",
+          namespace));
+    }
+  }
+
+  @Override
+  public List<NamespaceConfigurationDTO> findByIds(final List<Long> ids) {
+    return dao.get(ids);
+  }
+
+  @Override
+  public int delete(final NamespaceConfigurationDTO entity) {
+    return dao.delete(entity.getId());
+  }
+
+  @Override
+  public int deleteById(final Long id) {
+    return dao.delete(id);
+  }
+
+  @Override
+  public int deleteByIds(final List<Long> ids) {
+    return dao.delete(ids);
+  }
+
+  @Override
+  public int deleteByPredicate(final Predicate predicate) {
+    return dao.deleteByPredicate(predicate);
+  }
+
+  @Override
+  public int deleteRecordsOlderThanDays(final int days) {
+    final DateTime expireDate = new DateTime(DateTimeZone.UTC).minusDays(days);
+    final Timestamp expireTimestamp = new Timestamp(expireDate.getMillis());
+    final Predicate timestampPredicate = Predicate.LT("createTime", expireTimestamp);
+    return deleteByPredicate(timestampPredicate);
+  }
+
+  @Override
+  public List<NamespaceConfigurationDTO> findAll() {
+    return dao.getAll();
+  }
+
+  @Override
+  public List<NamespaceConfigurationDTO> findByPredicate(final Predicate predicate) {
+    return dao.get(predicate);
+  }
+
+  @Override
+  public List<NamespaceConfigurationDTO> filter(final DaoFilter daoFilter) {
+    return dao.filter(daoFilter);
+  }
+
+  @Override
+  public int update(final NamespaceConfigurationDTO entity, final Predicate predicate) {
+    return dao.update(entity, predicate);
+  }
+
+  @Override
+  public long count() {
+    return dao.count();
+  }
+
+  @Override
+  public long count(final Predicate predicate) {
+    return dao.count(predicate);
+  }
+
+  private TimeConfigurationDTO getDefaultTimeConfiguration() {
```

Comment:
this will break existing environments that have a custom time configuration.
to avoid breaking changes I think the default should be obtained from the Time configuration. 

once we are sure every users has migrated we remove the time configuration and hardcode it here

Reply:
Ah, right
this is a critical consideration. Let me fix this

Reply:
Done